pyproject.toml

```
@@ -1,11 +1,169 @@
+    "License :: OSI Approved :: Apache Software License",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Physics",
+]
+dependencies = [
+    "click>=8.0.0",  # for console scripts
+    "importlib_resources>=1.4.0; python_version < '3.9'",  # for resources in schema
+    "jsonpatch>=1.15",
+    "jsonschema>=4.15.0",  # for utils
+    "pyyaml>=5.1",  # for parsing CLI equal-delimited options
+    "scipy>=1.5.1",  # requires numpy, which is required by pyhf and tensorflow
+    "tqdm>=4.56.0",  # for readxml
+    "numpy",  # compatible versions controlled through scipy
+]
+
+[project.scripts]
+pyhf = "pyhf.cli:cli"
+
+[project.urls]
+Documentation = "https://pyhf.readthedocs.io/"
+Homepage = "https://github.com/scikit-hep/pyhf"
+"Issue Tracker" = "https://github.com/scikit-hep/pyhf/issues"
+"Release Notes" = "https://pyhf.readthedocs.io/en/stable/release-notes.html"
+"Source Code" = "https://github.com/scikit-hep/pyhf"
+
+# FIXME: Make recursive
+[project.optional-dependencies]
+shellcomplete = ["click_completion"]
+tensorflow = [
+    "tensorflow>=2.7.0",  # c.f. PR #1962
+    "tensorflow-probability>=0.11.0",  # c.f. PR #1657
+]
+torch = ["torch>=1.10.0"]  # c.f. PR #1657
+jax = [
+    "jax>=0.4.1",  # c.f. PR #2079
+    "jaxlib>=0.4.1",  # c.f. PR #2079
+]
+xmlio = ["uproot>=4.1.1"]  # c.f. PR #1567
+minuit = ["iminuit>=2.7.0"]  # c.f. PR #1895
+backends = ["pyhf[tensorflow,torch,jax,minuit]"]
+contrib = [
+    "matplotlib>=3.0.0",
+    "requests>=2.22.0",
+]
+all = ["pyhf[backends,xmlio,contrib,shellcomplete]"]
+
+test = [
+    "pyhf[all]",
+    "scikit-hep-testdata>=0.4.11",
+    "pytest>=6.0",
+    "coverage[toml]>=6.0.0",
+    "pytest-mock",
+    "requests-mock>=1.9.0",
+    "pytest-benchmark[histogram]",
+    "pytest-console-scripts",
+    "pytest-mpl",
+    "pydocstyle",
+    "papermill~=2.3.4",
+    "scrapbook~=0.5.0",
+    "jupyter",
+    "graphviz",
+    "pytest-socket>=0.2.0",  # c.f. PR #1917
+]
+
+docs = [
+    "pyhf[xmlio,contrib]",
+    "sphinx>=5.1.1",  # c.f. https://github.com/scikit-hep/pyhf/pull/1926
+    "sphinxcontrib-bibtex~=2.1",
+    "sphinx-click",
+    "sphinx_rtd_theme",
+    "nbsphinx!=0.8.8",  # c.f. https://github.com/spatialaudio/nbsphinx/issues/620
+    "ipywidgets",
+    "sphinx-issues",
+    "sphinx-copybutton>=0.3.2",
+    "sphinx-togglebutton>=0.3.0",
+    "ipython!=8.7.0",  # c.f. https://github.com/scikit-hep/pyhf/pull/2068
+]
+
+devleop = [
+    "pyhf[test,docs]",
+    "tbump>=6.7.0",
+    "pre-commit",
+    "nox",
+    "codemetapy>=2.3.0",
+]
+
+[tool.hatch]
+version.source = "vcs"
+build.hooks.vcs.version-file = "src/pyhf/_version.py"
```

I'd explode this, as it seems to be best practice.

```suggestion
[tool.hatch.version]
source = "vcs"

[tool.hatch.build.hooks.vcs]
version-file = "src/pyhf/_version.py"
```